### ISSUE TITLE
Failure to post to Kafka 

### ISSUE THREAD

**Issue by DmitryLitvintsev** [dCache/dcache#7272]
In the past I was complaining about failure to post to Kafka after upgrade to 8.2. 

I believe I finally got to the bottom of it. 

```
dcache.kafka.bootstrap-servers=lssrv08:9092
```

I think the issue is due to:

```
telnet -6 lssrv08 9092
Trying 2620:6a:0:8430:f0:0:240:59...
telnet: connect to address 2620:6a:0:8430:f0:0:240:59: Connection refused
```

vs 

```
 telnet -4 lssrv08 9092
Trying 131.225.240.59...
Connected to lssrv08.
Escape character is '^]'.
```

And I think the failure to connect to Kafka started to happen due to patch:

da994201

That added:

```
    -Djava.net.preferIPv6Addresses=system \
```

And indeed if I edit 

```
/usr/share/dcache/defaults/dcache.properties
```
and remove that line - things work. 

Question - is there convenient way to override/remove this property? 
Should I be redefining  the whole of `dcache.java.options` in `dcache.conf` ?

Thanks

**Comment by kofemann** [comment]
The problem is obviously in a network configuration that IPv4 and IPv6 have different firewall rules or IPv6 address points to a different host.

as a workaround, you can add `-Djava.net.preferIPv6Addresses=ipv4` to extra options:

```
dcache.java.options.extra=-Djava.net.preferIPv6Addresses=ipv4
```

As extra options added after standard option, the value will be overwritten

**Comment by DmitryLitvintsev** [comment]
yeah, I thought so too - I thought it was just a matter of firewall configuration and opened a ticket with "them" long ago. But then it turned almost insurmountable. The service runs on container and they cannot figure IPv6 for it looks like. Meanwhile I need data in ELK. 
Thanks for the tip. I will use it meanwhile!

**Comment by kofemann** [comment]
Actually, the valid values are true, false and system. All other values threated as false

**Comment by paulmillar** [comment]
Just FYI, in the past, `system` doesn't work with all JVMs.

(Also, I would say the "correct" solution is simply to remove the AAAA record from DNS, until the IPv6 deployment is fixed ;-)
